app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -183,69 +146,62 @@ class AddIncomeSourceStartDateController @Inject()(authenticate: AuthenticationP
-        )
-      case (false, UkProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKProperty,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showUKProperty
-        )
-      case (false, ForeignProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignProperty,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showForeignProperty
-        )
-      case (true, SelfEmployment) =>
-        (
-          controllers.incomeSources.add.routes.AddBusinessNameController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitSoleTraderBusinessAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showSoleTraderBusinessAgent
-        )
-      case (true, UkProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKPropertyAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showUKPropertyAgent
-        )
-      case (true, ForeignProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignPropertyAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showForeignPropertyAgent
+    (startDateKey, isUpdate) match {
+      case (Some(key), true) if Try(LocalDate.parse(key)).toOption.isDefined =>
+        form.fill(
+          DateFormElement(
+            LocalDate.parse(key)
+          )
         )
+      case _ => form
     }
   }
 
-  def changeBusinessStartDate(): Action[AnyContent] = Action(Ok("Change Business Start Date WIP"))
-  def changeBusinessStartDateAgent(): Action[AnyContent] = Action(Ok("Agent Change Business Start Date WIP"))
-  def changeUKPropertyStartDate(): Action[AnyContent] = Action(Ok("Change UK Property Start Date WIP"))
-  def changeUKPropertyStartDateAgent(): Action[AnyContent] = Action(Ok("Agent Change UK Property Start Date WIP"))
-  def changeForeignPropertyStartDate(): Action[AnyContent] = Action(Ok("Change Foreign Property Start Date WIP"))
-  def changeForeignPropertyStartDateAgent(): Action[AnyContent] = Action(Ok("Agent Change Foreign Property Start Date WIP"))
+  private def getCalls(incomeSourceType: IncomeSourceType,
+                       isAgent: Boolean,
+                       isUpdate: Boolean): (Call, Call, Call) = {
+    (
+      (isAgent, isUpdate, incomeSourceType) match {
```

Comment:
assuming this is finite mapping / one to one, I would suggest to use map instead. ie. you can refactor your code to something like:
`
  private val agentCalls: Map[IncomeSourceType, Call] = Map(
    SelfEmployment -> routes.AddBusinessNameController.showAgent(),
    UkProperty -> routes.CheckUKPropertyDetailsController.showAgent(),
    ForeignProperty -> routes.ForeignPropertyCheckDetailsController.showAgent()
  )

  
  private val individualCalls: Map[IncomeSourceType, Call] = Map(
    SelfEmployment -> routes.AddBusinessNameController.showAgent(),
    UkProperty -> routes.CheckUKPropertyDetailsController.showAgent(),
    ForeignProperty -> routes.ForeignPropertyCheckDetailsController.showAgent()
  )

  private def getCalls(incomeSourceType: IncomeSourceType,
                       isAgent: Boolean,
                       isUpdate: Boolean): (Call, Call, Call) = {
    (
      if (isAgent)
         agentCalls(incomeSourceType)
      else
        individualCalls(incomeSourceType)
     ,
      routes.AddIncomeSourceStartDateController.submit(incomeSourceType.key, isAgent, isUpdate),
      routes.AddIncomeSourceStartDateCheckController.show(incomeSourceType.key, isAgent, isUpdate)
    )
  }
`